src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -345,712 +356,129 @@ private void storeMergedFeed() throws IOException, CheckedAWSException {
-            // MTC requires that the stop and route records be merged based on different key fields.
-            switch (table.name) {
-                case "stops":
-                    keyField = "stop_code";
-                    break;
-                case "routes":
-                    keyField = "route_short_name";
-                    break;
-                default:
-                    // Otherwise, use the standard key field (see keyField declaration.
-                    break;
-            }
-        }
-        // Set up objects for tracking the rows encountered
-        Map<String, String[]> rowValuesForStopOrRouteId = new HashMap<>();
-        Set<String> rowStrings = new HashSet<>();
-        // Track shape_ids found in future feed in order to check for conflicts with active feed (MTC only).
-        Set<String> shapeIdsInFutureFeed = new HashSet<>();
-        int mergedLineNumber = 0;
-        // Get the spec fields to export
-        List<Field> specFields = table.specFields();
-        boolean stopCodeMissingFromFirstFeed = false;
+        MergeLineContext ctx = new MergeLineContext(this, table, out);
         try {
-            // Get shared fields between all feeds being merged. This is used to filter the spec fields so that only
-            // fields found in the collection of feeds are included in the merged table.
-            Set<Field> sharedFields = getSharedFields(feedsToMerge, table);
-            // Initialize future feed's first date to the first calendar date from the validation result.
-            // This is equivalent to either the earliest date of service defined for a calendar_date record or the
-            // earliest start_date value for a calendars.txt record. For MTC, however, they require that GTFS
-            // providers use calendars.txt entries and prefer that this value (which is used to determine cutoff
-            // dates for the active feed when merging with the future) be strictly assigned the earliest
-            // calendar#start_date (unless that table for some reason does not exist).
-            LocalDate futureFeedFirstDate = feedsToMerge.get(0).version.validationResult.firstCalendarDate;
-            LocalDate futureFirstCalendarStartDate = LocalDate.MAX;
             // Iterate over each zip file. For service period merge, the first feed is the future GTFS.
             for (int feedIndex = 0; feedIndex < feedsToMerge.size(); feedIndex++) {
-                boolean keyFieldMissing = false;
-                // Use for a new agency ID for use if the feed does not contain one. Initialize to
-                // null. If the value becomes non-null, the agency_id is missing and needs to be
-                // replaced with the generated value stored in this variable.
-                String newAgencyId = null;
+                ctx.startNewFeed(feedIndex);
                 mergeFeedsResult.feedCount++;
```

Comment:
This `feedCount` field is never read from anywhere. Please remove it.

Reply:
Removed in https://github.com/ibi-group/datatools-server/pull/376/commits/e90b8c1e126267f0984a139725010523284e9136.